apps/ledger-live-mobile/src/components/GradientContainer.tsx

```diff
@@ -1,40 +1,71 @@
-import React from "react";
+import React, { useCallback } from "react";
 import Svg, { Defs, LinearGradient, Rect, Stop } from "react-native-svg";
 import { useTheme } from "styled-components/native";
 import { Flex } from "@ledgerhq/native-ui";
 import { StyleProp, ViewStyle } from "react-native";
 
+type Direction = "top-to-bottom" | "left-to-right" | "bottom-to-top" | "right-to-left";
 type Props = {
   color?: string;
   containerStyle?: StyleProp<ViewStyle>;
   gradientStyle?: StyleProp<ViewStyle>;
   children?: React.ReactNode;
   startOpacity?: number;
   endOpacity?: number;
+  direction?: Direction;
 };
 
+function getPositions(direction: Direction) {
+  let x1 = "0%";
+  let y1 = "0%";
+  let x2 = "0%";
+  let y2 = "100%";
+
+  switch (direction) {
+    case "left-to-right":
+      x2 = "100%";
+      y2 = "0%";
+      break;
+    case "bottom-to-top":
+      y1 = "100%";
+      y2 = "0%";
+      break;
+    case "right-to-left":
+      x1 = "100%";
+      x2 = "0%";
+      break;
+  }
+
+  return {
+    x1,
+    x2,
+    y1,
+    y2,
+  };
+}
+
 export default function GradientContainer({
   color,
   containerStyle,
   gradientStyle,
   children,
   startOpacity = 1,
   endOpacity = 0,
+  direction = "top-to-bottom",
 }: Props) {
   const { colors } = useTheme();
 
+  const positions = useCallback(() => getPositions(direction), [direction]);
```

Comment:
We don't need the `useCallback` here. I think that we can directly do the following:
`const gradientPositions = getPositions(direction);`
If you really have a re-rendering issue, favor useMemo to keep the same object.
Here we keep in memory the function but we always have a new object for `gradientPositions`, so we don't benefit from memorization + as we spread the object in the `LinearGradient` React component, React should be smart enough to compare the string for each value and do not perform the rerender.